src/Parser/Parser.hs

```diff
@@ -360,35 +364,50 @@ xtype = (try $ do
 -- | Parse just a type, before it is possibly extended
 xtype' :: Parser Xtype
 xtype' =
-  (try $ capIdentifier >>= lookupSyn)
+  (try typeName)
   <|>
   (try $ X <$> (pure Top) <*> enum) -- Plain enum
   <|>
   (try $ X <$> btype <*> (pure S.empty))
   <|>
   (try $ Tup <$> parens (lexeme ((:) <$> (xtype <* comma) <*> (commaSep1 xtype))))
 
+-- | Parse an xtype, but only if it is one of the base types or it has been declared
+--   and named (with the type keyword)
+namedType :: Parser Xtype
+namedType =
+   (try $ notExtended typeName)
+   <|>
+   (try $ Tup <$> parens (lexeme ((:) <$> (namedType <* comma) <*> (commaSep1 namedType))))
+   <|>
+   (notExtended $ X <$> btype <*> pure S.empty)
+   <?>
+   "a pre-existing or previously-declared type"
```

Comment:
Any suggestions for a better error message?

Reply:
Running this locally for the following program:
```haskell
game Test
test : QA
test = 9
```
I get:
```
Language Error: "Program" (line 4, column 1):
unexpected "t"
expecting a pre-existing or previously-declared type
Type QA not found!
```
It's more or less to the point...but I wish it were just a couple of lines, like `unexpected 't'` followed by `got 'QA', but expected a pre-existing or previously-declared type`.

Reply:
This is bad, thanks for the example. In particular, the `unexpected "t"` is not great. Anyways, the "expected a pre-exisiting..." and "Type QA..." messages are produced by two different parsers. I can fiddle around with it a little more to see if I can improve it.

Reply:
I could not get rid of the "unexpected t"; [other parsec users have had issues with some spurious error messages](https://github.com/haskell/parsec/issues/91) (however, I am not saying that it's not our fault). Note that my changes did not introduce it, you can see the same "unexpected t" message in the current version of bogl without these changes. 

As for your proposed error reword, I think it is important to indicate that QA was not declared (as opposed to just "got QA"). Then the message would be `Type QA was not declared; expecting a...` which is not too different from my latest change:

```
Language Error: "Program" (line 4, column 1):
unexpected "t"
expecting a base type or previously-declared type
Type QA not declared!
```

I also had difficulty displaying the message you are requesting while maintaining generality for the other cases (I am not too great with parsec, I'll admit). If you have a better approach, I am all for it.

Fortunately, other related error messages are better:

```
test : Int & {A}
test = 1
```

produces

```
Language Error: "Program" (line 3, column 13):
unexpected "&"
expecting a base type or previously-declared type
```